src/microstrain_3dm.cpp

```diff
@@ -0,0 +1,3381 @@
+    u16 device_descriptors_size  = 128*2;
+    u8  gps_source     = 0;
+    u8  heading_source = 0x1;
+    mip_low_pass_filter_settings filter_settings;
+    u16 duration = 0;
+    mip_filter_external_gps_update_command external_gps_update;
+    mip_filter_external_heading_update_command external_heading_update;
+    mip_filter_external_heading_with_time_command external_heading_with_time;
+
+    com_mode = 0;
+
+    //Device model flags
+    GX5_15 = false;
+    GX5_25 = false;
+    GX5_35 = false;
+    GX5_45 = false;
+
+    // ROS setup
+    ros::Time::init();
+    ros::NodeHandle node;
+    ros::NodeHandle private_nh("~");
+
+    // ROS Parameters
+    // Comms Parameters
+    std::string port;
+    int baud, pdyn_mode;
+    private_nh.param("port", port, std::string("/dev/ttyACM0"));
+    private_nh.param("baudrate",baud,115200);
+    baudrate = (u32)baud;
+    // Configuration Parameters
+    private_nh.param("device_setup",device_setup,false);
+    private_nh.param("readback_settings",readback_settings,true);
+    private_nh.param("save_settings",save_settings,true);
+
+    private_nh.param("auto_init",auto_init,true);
+    private_nh.param("gps_rate",gps_rate_, 1);
+    private_nh.param("imu_rate",imu_rate_, 10);
+    private_nh.param("nav_rate",nav_rate_, 10);
+    private_nh.param("dynamics_mode",pdyn_mode,1);
+    dynamics_mode = (u8)pdyn_mode;
+    if (dynamics_mode < 1 || dynamics_mode > 3){
+      ROS_WARN("dynamics_mode can't be %#04X, must be 1, 2 or 3.  Setting to 1.",dynamics_mode);
+      dynamics_mode = 1;
+    }
+    private_nh.param("declination_source",declination_source,2);
+    if (declination_source < 1 || declination_source > 3){
+      ROS_WARN("declination_source can't be %#04X, must be 1, 2 or 3.  Setting to 2.",declination_source);
+      declination_source = 2;
+    }
+    declination_source_u8 = (u8)declination_source;
+
+    //declination_source_command=(u8)declination_source;
+    private_nh.param("declination",declination,0.23);
+    private_nh.param("gps_frame_id",gps_frame_id_, std::string("wgs84"));
+    private_nh.param("imu_frame_id",imu_frame_id_, std::string("base_link"));
+    private_nh.param("odom_frame_id",odom_frame_id_, std::string("wgs84"));
+    private_nh.param("odom_child_frame_id",odom_child_frame_id_,
+		     std::string("base_link"));
+
+    private_nh.param("publish_imu",publish_imu_, true);
+
+    private_nh.param("publish_bias",publish_bias_, true);
+
+    // ROS publishers and subscribers
+    if (publish_imu_)
+      imu_pub_ = node.advertise<sensor_msgs::Imu>("imu/data",100);
+
+
+    //Publishes device status
+    device_status_pub_ = node.advertise<microstrain_3dm::status_msg>("device/status", 100);
+
+
+    //Services to set/get device functions
+    ros::ServiceServer service = node.advertiseService("reset_kf", &Microstrain::reset_callback, this);
+    ros::ServiceServer service3 = node.advertiseService("DeviceReport", &Microstrain::device_report, this);
```

Comment:
Can the service name be switched to all lower case with underscores to match the ROS style.